pkg/apis/workspaces/v1alpha1/pluginLikeComponents.go

```diff
@@ -26,19 +26,19 @@ type ChePluginLocation struct {
 	// +optional
 	LocationType PluginLocationType `json:"locationType"`
 
-	// Location in a plugin registry
+	// Location of an entry inside a plugin registry
 	// +optional
-	Registry *RegistryLocation `json:"registry,omitempty"`
+	RegistryEntry *RegistryEntryPluginLocation `json:"registryEntry,omitempty"`
 
-	// Location defined as an URL
+	// Location defined as an URI
 	// +optional
-	Url string `json:"url,omitempty"`
+	Uri string `json:"uri,omitempty"`
 }
 
 type PluginLikeComponent struct {
 	BaseComponent `json:",inline"`
 	MemoryLimit  string `json:"memoryLimit,omitempty"`
-	Location ChePluginLocation `json:",inline"`
+	ChePluginLocation `json:",inline"`
```

Comment:
Don't know why you suggest this. Since it should be inline here it should not have the `location` json field name.
Having it inline is important to have it correctly managed (unmarshalled) as a Kubernetes union.

Reply:
Sorry, I don't know either. Don't mind.